github/actions_secrets_test.go

```diff
@@ -0,0 +1,128 @@
+	})
+
+	key, _, err := client.Actions.GetPublicKey(context.Background(), "o", "r")
+	if err != nil {
+		t.Errorf("Actions.GetPublicKey returned error: %v", err)
+	}
+
+	want := &PublicKey{ID: String("1234"), Key: String("2Sg8iYjAxxmI2LvUXpJjkYrMxURPc8r+dB7TJyvv1234")}
+	if !reflect.DeepEqual(key, want) {
+		t.Errorf("Actions.GetPublicKey returned %+v, want %+v", key, want)
+	}
+}
+
+func TestActionsService_ListSecrets(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/secrets", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testFormValues(t, r, values{"page": "2"})
+		fmt.Fprint(w, `{"total_count":2,"secrets":[{"name":"A","created_at":"2019-01-02T15:04:05Z","updated_at":"2020-01-02T15:04:05Z"},{"name":"B","created_at":"2019-01-02T15:04:05Z","updated_at":"2020-01-02T15:04:05Z"}]}`)
+	})
+
+	opt := &ListOptions{Page: 2}
+	secrets, _, err := client.Actions.ListSecrets(context.Background(), "o", "r", opt)
+	if err != nil {
+		t.Errorf("Actions.ListSecrets returned error: %v", err)
+	}
+
+	want := []*Secret{
+		{Name: String("A"), CreatedAt: &Timestamp{time.Date(2019, time.January, 02, 15, 04, 05, 0, time.UTC)}, UpdatedAt: &Timestamp{time.Date(2020, time.January, 02, 15, 04, 05, 0, time.UTC)}},
+		{Name: String("B"), CreatedAt: &Timestamp{time.Date(2019, time.January, 02, 15, 04, 05, 0, time.UTC)}, UpdatedAt: &Timestamp{time.Date(2020, time.January, 02, 15, 04, 05, 0, time.UTC)}},
+	}
+	if !reflect.DeepEqual(secrets, want) {
+		t.Errorf("Actions.ListSecrets returned %+v, want %+v", secrets, want)
+	}
+}
+
+func TestActionsService_GetSecret(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/secrets/NAME", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		fmt.Fprint(w, `{"name":"NAME","created_at":"2019-01-02T15:04:05Z","updated_at":"2020-01-02T15:04:05Z"}`)
+	})
+
+	secret, _, err := client.Actions.GetSecret(context.Background(), "o", "r", "NAME")
+	if err != nil {
+		t.Errorf("Actions.GetSecret returned error: %v", err)
+	}
+
+	want := &Secret{
+		Name:      String("NAME"),
+		CreatedAt: &Timestamp{time.Date(2019, time.January, 02, 15, 04, 05, 0, time.UTC)},
+		UpdatedAt: &Timestamp{time.Date(2020, time.January, 02, 15, 04, 05, 0, time.UTC)},
+	}
+	if !reflect.DeepEqual(secret, want) {
+		t.Errorf("Actions.GetSecret returned %+v, want %+v", secret, want)
+	}
+}
+
+func TestActionsService_CreateOrUpdateSecret(t *testing.T) {
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/actions/secrets/NAME", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "PUT")
+		testHeader(t, r, "Content-Type", "application/json")
+		testBody(t, r, `{"key_id":"1234","encrypted_value":"QIv="}`+"\n")
+		fmt.Fprint(w, `{"key_id":"1234","encrypted_value":"QIv="}`)
```

Comment:
This endpoint will not return a response... you can remove this and only set the return code to 201 or 204.
(`http.StatusCreated` or `http.StatusNoContent`)

Reply:
Done